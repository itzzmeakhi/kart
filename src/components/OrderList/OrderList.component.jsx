import React from 'react';
import { connect } from 'react-redux';

import Order from './../Order/Order.component';

import './OrderList.styles.scss';

const OrderList = ({ userOrders }) => {
    return(
        <div className = 'order__list'>
            {userOrders.map((ord, index) => {
                return <Order key = {`order${index}`} orderData = {ord} />
            })}
        </div>
    )
}

const mapStateToProps = state => {
    return {
        userOrders: state.user.loggedInUser.userOrders
    }
}

export default connect(mapStateToProps)(OrderList);