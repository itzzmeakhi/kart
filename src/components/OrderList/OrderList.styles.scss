@import './../../scss/colors';

.order__list {
    width: 90%;
    display: flex;
    flex-direction: column;
    background: $color-white;
    margin: 0 auto;
    > * {
        border: 2px solid $color-white-thick;
        margin: 10px;
        color: $color-blue-thick;
    }
}