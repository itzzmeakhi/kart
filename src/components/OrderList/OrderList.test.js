import React from 'react';
import { shallow } from 'enzyme';

import OrderList from './OrderList.component';
import { storeFactory } from './../../test.utils';


const setup = (initialState={}) => {
    const store = storeFactory(initialState);
    const wrapper = shallow(<OrderList store = {store} />).dive();
    return wrapper;
}


describe('<OrderList />', () => {
    let wrapper;
    beforeEach(() => {
        const state = {
            user: {
                loggedInUser: {
                    userOrders: []
                }
            }
        }
        wrapper = setup(state);
    })

    it('should component rendered without errors', () => {
        expect(wrapper.length).toBe(1);
    })

    it('should have valid props passed', () => {
        expect(wrapper.props().userOrders).toEqual([]);
    })

    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

})