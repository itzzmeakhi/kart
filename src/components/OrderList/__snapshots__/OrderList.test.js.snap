// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderList /> should match the snapshot 1`] = `ShallowWrapper {}`;
